Scripts/test.sh

```diff
@@ -23,18 +23,38 @@ struct Secrets {
 }
 EOF
 
+# We want to specify a custom derived data directory so that we know where to
+# find the test result .xcresult bundle. However, it appears that putting the
+# derived data directory inside a Swift package’s repo causes build errors (see
+# https://forums.swift.org/t/xcode-and-swift-package-manager/44704/6). So I’m
+# following the suggestion there of placing the derived data directory outside
+# the repo.
+temp_dir=`mktemp -d -t ably-asset-tracking-swift`
+derived_data_dir="${temp_dir}/DerivedData"
+
 # If xcodebuild fails (e.g. due to failed tests), we want to defer the failure
-# of this script until we’ve had a chance to copy the test results JUnit report
-# to the place where test-observability-action expects it to be. Hence we
-# temporarily disable the -e option.
+# of this script until we’ve had a chance to process and copy the logs that
+# xcodebuild created. Hence we temporarily disable the -e option.
 set +e
-# --report: "Creates a JUnit-style XML report at build/reports/junit.xml"
-set -o pipefail && xcodebuild test -scheme "ably-asset-tracking-swift-Package" -destination 'platform=iOS Simulator,name=iPhone 12' \
-	| xcpretty --report junit
+set -o pipefail && xcodebuild test -scheme "ably-asset-tracking-swift-Package" -destination 'platform=iOS Simulator,name=iPhone 12' -derivedDataPath "${derived_data_dir}" \
+	| xcpretty
 xcodebuild_exit_status=$?
 set -e
 
-# test-observability-action looks for .junit files
-cp build/reports/junit.xml results.junit
+# Create a directory to store all of the files we’ll pass to actions
```

Comment:
I'm not sure "pass to actions" is the best turn of phrase, especially for those with less experience with GitHub's ecosystem. Perhaps:

```suggestion
# Create a directory to store all of the files we wish to persist outside of the GitHub
# workflow that called this script. That workflow uses the upload-artifact action to
# archive and save the contents of this folder for later inspection.
```

Reply:
Good point. I've replaced it with the following:

```
# Create a directory to store all of the files that we wish to pass to
# subsequent actions in the GitHub workflow that called this script.
```

I didn't want to get too verbose, since it isn't just upload-artifact that we use – there's also test-observability-action.

I think this still keeps to the spirit of your suggested change, though?

Reply:
Nice. Thanks!